firmware/defmt-test/macros/tests/ui/teardown-has-invalid-function-signature.stderr

```diff
@@ -0,0 +1,5 @@
+error: `#[teardown]` function must have signature `fn(state: &mut Type)` (parameter is optional)
```

Comment:
maybe we can change this to:
```suggestion
error: `#[teardown]` function must have signature `fn()` or `fn(state: &mut T)`
```

Reply:
I copied the message from before_each. after_each and test also have the same message. should I change all of them?

Reply:
yes. if we use the same message (in parts) in all of them maybe we can store them in a constant?

Reply:
I went with a macro to concatenate the string literals.

The concat!() macro only works with string literals. That‘s why I chose the macro instead of a constant.